check.sh

```diff
@@ -34,11 +34,14 @@ Commands:
 Options:
     -h, --help    print this help text
     --double      run check with double-precision
+    -f, --filter <args>     only run integration tests which contain any of the
+                            args (comma-separated). requires itest.
 
 Examples:
     check.sh fmt clippy
     check.sh
     check.sh --double clippy
+    chech.sh test itest -f variant ffi
```

Comment:
Could you make the implementation only accept comma-separated values, as I suggested in the help text?
```
    --filter <args>    only run integration tests which contain any of the
                       args (comma-separated). requires itest.
```


Otherwise you get the ambiguity of ordering, which can be unintuitive.
```suggestion
    chech.sh test itest -f variant,ffi
```
It should also simplify implementation that you would only need to consider the immediate next argument, and not any arguments following `--filter`.